lib/rules/no-unsupported-role-attributes.js

```diff
@@ -0,0 +1,84 @@
+import { roles, elementRoles } from 'aria-query';
+
+import AstNodeInfo from '../helpers/ast-node-info.js';
+import Rule from './_base.js';
+
+function createUnsupportedAttributeErrorMessage(attr, role, element) {
+  if (element) {
+    return `The attribute ${attr} is not supported by the element ${element} with the implicit role of ${role}`;
+  } else {
+    return `The attribute ${attr} is not supported by the role ${role}`;
+  }
+}
+function getImplicitRole(element, typeAttribute) {
+  if (element === 'input') {
+    for (let key of elementRoles.keys()) {
+      if (key.name === element) {
+        let attributes = key.attributes;
+        for (let attribute of attributes) {
+          if (attribute.name === 'type' && attribute.value === typeAttribute) {
+            return elementRoles.get(key)[0];
+          }
+        }
+      }
+    }
+  } else {
```

Comment:
Can we use an early return instead here?

I found a lint rule that I think would enforce this: https://github.com/Shopify/web-configs/blob/main/packages/eslint-plugin/docs/rules/prefer-early-return.md

Reply:
Yes thank you for pointing this out, the code has been updated with an early return.